scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -522,6 +532,42 @@
   (autoscaling? [_ pool-name]
     (and (-> synthetic-pods-config :pools (contains? pool-name))
          (= @state-atom :running)))
+  
+  (max-launchable
+   [this pool-name]
+   (let [scheduling-pods (get-scheduling-pods this pool-name)
+         num-scheduling-pods (count scheduling-pods)
+         total-pods (-> @all-pods-atom keys count)
+         total-nodes (-> @current-nodes-atom keys count)
+         ; NOTE: the synthetic pods config is co-opted here to determine the max outstanding
+         ; pods in the compute cluster. This configuration, set on the compute cluster
+         ; template, will be refactored in the future to more generally define autoscaling 
+         ; parameters once synthetic pods are decommissioned.
+         {:keys [max-pods-outstanding max-total-pods max-total-nodes]
```

Comment:
Why not default max-pods-outstanding here too? (The concern if that's not in synthetic-pods-config, this will blow up with an NPE with.... impressive blast radius.)

Conversely, shouldn't this all be defaulted in synthetic-pods-config, or the factory method for this defrecord? (I'd prefer that to be honest.

Reply:
Done, I've made max total nodes and pods required as part of the guard below. Additionally, confirmed out current prod configs define these.